python/examples/explore.py

```diff
@@ -0,0 +1,295 @@
+# as an argument (`featurize`) to `chemiscope.explore`. Inside this function, we perform
+# descriptor calculation using SOAP and then reduce the dimensionality with `Kernel PCA
+# <https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.KernelPCA.html/>`_.
+#
+# First, let's import the necessary packages.
+from dscribe.descriptors import SOAP  # noqa
+from sklearn.decomposition import KernelPCA  # noqa
+
+# %%
+#
+# Define the function `soap_kpca` which takes one argument (`frames`). This argument
+# contains the structures provided to `chemiscope.explore` and is internally passed
+# to the `featurize` function.
+
+
+def soap_kpca(frames):
+    # Initialise soap calculator. The detailed explanation of the provided
+    # hyperparameters can be checked in the documentation of the library (`dscribe`).
+    soap = SOAP(
+        # the dataset used in the example contains only carbon
+        species=["C"],
+        r_cut=4.5,
+        n_max=8,
+        l_max=6,
+        sigma=0.2,
+        rbf="gto",
+        average="outer",
+        periodic=True,
+        weighting={"function": "pow", "c": 1, "m": 5, "d": 1, "r0": 3.5},
+    )
+
+    # Compute features
+    descriptors = soap.create(frames)
+
+    # Apply KPCA
+    transformer = KernelPCA(n_components=2, gamma=0.05)
+
+    # Return a 2D array of reduced features
+    return transformer.fit_transform(descriptors)
+
+
+# %%
+#
+# Provide the created function to `chemiscope.explore`.
+
+cs = chemiscope.explore(frames, featurize=soap_kpca)
+
+# %%
+#
+# Note: It is possible to add parallelization when computing the SOAP descriptors and
+# performing dimentionality reduction with KernelPCA by providing the `n_jobs`
+# parameter. This allows the computation to utilize multiple CPU cores for faster
+# processing. An example of how to include `n_jobs` is shown below on this page.
+#
+# To showcase the results of the `soap_kpca` function, we have pre-computed it for
+# the 6k structures from the C-GAP-20U dataset. You can fetch and visualize this
+# pre-computed dimensionality reduction as follows:
+
+# sphinx_gallery_start_ignore
+import requests  # noqa
+
+
+def fetch_dataset(filename, base_url="https://zenodo.org/records/12626972/files/"):
+    """Helper function to load the pre-computed examples"""
+    local_path = "data/" + filename
+    if not os.path.isfile(local_path):
+        response = requests.get(base_url + filename)
+        with open(local_path, "wb") as file:
+            file.write(response.content)
+
+
+# sphinx_gallery_end_ignore
+
+fetch_dataset("soap_kpca_c-gap-20u.json.gz")
+chemiscope.show_input("data/soap_kpca_c-gap-20u.json.gz")
+
+
+# %%
+#
+# Example with MACE-OFF and t-SNE
+# +++++++++++++++++++++++++++++++++++++
+#
+# In this part, we are going to define another `featurize` function that runs
+# calculation of desciptors with MACE-OFF and uses t-SNE for the dimensionality
+# reduction.
```

Comment:
can you link to both MACE and t-SNE documentation here? Not everyone will be familiar with these